lib/docs/scrapers/docker.rb

```diff
@@ -0,0 +1,42 @@
+module Docs
+  class Docker < UrlScraper
+    self.name = 'Docker'
+    self.slug = 'docker'
+    self.type = 'docker'
+
+    self.links = {
+      home: 'https://docker.com/',
+      code: 'https://github.com/docker/docker'
+    }
+
+    html_filters.push 'docker/clean_html', 'docker/entries'
+
+    options[:trailing_slash] = true
+
+    options[:only_patterns] = [
+      /\Aengine\//,
+      /\Acompose\//,
+      /\Amachine\//,
+      /\Aswarm\//
+    ]
+
+    options[:skip] = [
+      "engine/installation/ubuntulinux/" # invalid document
```

Comment:
I think this is a bug in the document parser. Instead of responding with a 301/302, docker.com instead responds with a "valid" web page that triggers a redirect via an HTML header. This line is just here so the scraper doesn't crash.